frontend/src/component/common/PrettifyLargeNumber/PrettifyLargeNumber.tsx

```diff
@@ -0,0 +1,45 @@
+import { FC } from 'react';
+import millify from 'millify';
+import { Tooltip } from '@mui/material';
+import { LARGE_NUMBER_PRETTIFIED } from 'utils/testIds';
+
+interface IPrettifyLargeNumberProps {
+    /**
+     * Value to prettify
+     */
+    value: number;
+    /**
+     * Threshold above which the number will be prettified. Values lower than this will just have comma separators added
+     * @default 1_000_000
+     */
+    threshold?: number;
+    /**
+     * The number of significant figures
+     * @default 2
+     */
+    precision?: number;
+}
+
+export const PrettifyLargeNumber: FC<IPrettifyLargeNumberProps> = ({
+    value,
+    threshold = 1_000_000,
+    precision = 2,
+}) => {
+    let prettyValue: string;
+    let showTooltip = false;
+
+    if (value < threshold) {
+        prettyValue = value.toLocaleString();
+    } else {
+        prettyValue = millify(value, { precision });
+        showTooltip = true;
+    }
+
+    return showTooltip ? (
+        <Tooltip title={value.toLocaleString()} arrow>
+            <span data-testid={LARGE_NUMBER_PRETTIFIED}>{prettyValue}</span>
+        </Tooltip>
+    ) : (
+        <span data-testid={LARGE_NUMBER_PRETTIFIED}>{prettyValue}</span>
+    );
```

Comment:
We have a `ConditionallyRender` component we like to use for these cases, would make this more consistent with the rest of the code base:

```JS
return (
  <ConditionallyRender
      condition={showTooltip}
      show={
          <Tooltip title={value.toLocaleString()} arrow>
              <span data-testid={LARGE_NUMBER_PRETTIFIED}>
                  {prettyValue}
              </span>
          </Tooltip>
      }
      elseShow={
          <span data-testid={LARGE_NUMBER_PRETTIFIED}>{prettyValue}</span>
      }
  />
);
```

Maybe there's also an opportunity for more DRYness on the `span`:

```JS
const valueSpan = (
  <span data-testid={LARGE_NUMBER_PRETTIFIED}>{prettyValue}</span>
);
```